pkgs/applications/networking/instant-messengers/viber/default.nix

```diff
@@ -1,111 +1,28 @@
-{fetchurl, lib, stdenv, dpkg, makeWrapper,
- alsa-lib, cups, curl, dbus, expat, fontconfig, freetype, glib, gst_all_1,
- harfbuzz, libcap, libGL, libGLU, libpulseaudio, libxkbcommon, libxml2, libxslt,
- nspr, nss, openssl_1_1, systemd, wayland, xorg, zlib, ...
-}:
-
-stdenv.mkDerivation {
-  pname = "viber";
-  version = "16.1.0.37";
-
-  src = fetchurl {
-    # Official link: https://download.cdn.viber.com/cdn/desktop/Linux/viber.deb
-    url = "https://web.archive.org/web/20211119123858/https://download.cdn.viber.com/cdn/desktop/Linux/viber.deb";
-    sha256 = "sha256-hOz+EQc2OOlLTPa2kOefPJMUyWvSvrgqgPgBKjWE3p8=";
-  };
-
-  nativeBuildInputs = [ makeWrapper ];
-  buildInputs = [ dpkg ];
-
-  dontUnpack = true;
-
-  libPath = lib.makeLibraryPath [
-      alsa-lib
-      cups
-      curl
-      dbus
-      expat
-      fontconfig
-      freetype
-      glib
-      gst_all_1.gst-plugins-base
-      gst_all_1.gstreamer
-      harfbuzz
-      libcap
-      libGLU libGL
-      libpulseaudio
-      libxkbcommon
-      libxml2
-      libxslt
-      nspr
-      nss
-      openssl_1_1
-      stdenv.cc.cc
-      systemd
-      wayland
-      zlib
-
-      xorg.libICE
-      xorg.libSM
-      xorg.libX11
-      xorg.libxcb
-      xorg.libXcomposite
-      xorg.libXcursor
-      xorg.libXdamage
-      xorg.libXext
-      xorg.libXfixes
-      xorg.libXi
-      xorg.libXrandr
-      xorg.libXrender
-      xorg.libXScrnSaver
-      xorg.libXtst
-      xorg.xcbutilimage
-      xorg.xcbutilkeysyms
-      xorg.xcbutilrenderutil
-      xorg.xcbutilwm
-  ]
-  ;
-
-  installPhase = ''
-    dpkg-deb -x $src $out
-    mkdir -p $out/bin
-
-    # Soothe nix-build "suspicions"
-    chmod -R g-w $out
-
-    for file in $(find $out -type f \( -perm /0111 -o -name \*.so\* \) ); do
-      patchelf --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" "$file" || true
-      patchelf --set-rpath $libPath:$out/opt/viber/lib $file || true
-    done
-
-    # qt.conf is not working, so override everything using environment variables
-    wrapProgram $out/opt/viber/Viber \
-      --set QT_QPA_PLATFORM "xcb" \
-      --set QT_PLUGIN_PATH "$out/opt/viber/plugins" \
-      --set QT_XKB_CONFIG_ROOT "${xorg.xkeyboardconfig}/share/X11/xkb" \
-      --set QTCOMPOSE "${xorg.libX11.out}/share/X11/locale" \
-      --set QML2_IMPORT_PATH "$out/opt/viber/qml"
-    ln -s $out/opt/viber/Viber $out/bin/viber
-
-    mv $out/usr/share $out/share
-    rm -rf $out/usr
-
-    # Fix the desktop link
-    substituteInPlace $out/share/applications/viber.desktop \
-      --replace /opt/viber/Viber $out/opt/viber/Viber \
-      --replace /usr/share/ $out/share/
+{ callPackage, buildFHSEnv }:
+let
+  unwrapped = callPackage ./unwrapped.nix { };
+in
+buildFHSEnv
```

Comment:
Which `patchelf` are you referring to? The one in the unwrapped package? I find that the `patchelf` helps greatly to actually find the missing libraries because it goes through all the bundled libraries too. 

I initially tried to avoid using `buildFHSEnv` but unfortunately the bundled Qt library version is 6.5 and there have been various issues with the 6.6 version that is currently available in unstable, so I had to use the bundled one which looks into FHS paths for just a few dependencies. 

I've since been in contact with the Qt maintainers for the issues and hopefully I will be able to purge the bundled dependencies once the patches are in.

Reply:
Could you give me some references? What time frame are we talking for the patches?

Reply:
The application /seems/ to fall victim of [QTBUG-117899](https://bugreports.qt.io/browse/QTBUG-117899).
I also reported an ABI break which affects the application, see the [relevant codereview.](https://codereview.qt-project.org/c/qt/qtdeclarative/+/468501/11/src/qml/qml/qqmlprivate.h#706)